dotnet/src/Agents/Abstractions/KernelAgent.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Base class for agents utilizing <see cref="Microsoft.SemanticKernel.Kernel"/> plugins or services.
+/// </summary>
+public abstract class KernelAgent : Agent
+{
+    /// <summary>
+    /// The arguments used to optionally format <see cref="Instructions"/>.
+    /// </summary>
+    public KernelArguments? InstructionArguments { get; set; }
```

Comment:
Do we need this in the initial implementation?

Do we expect the instruction arguments to change over the lifetime of the agent?

I would favour keeping this simple to start with and just passing in the instructions string and if a developer want to generate this from a prompt template they do this once up front.